app/group.go

```diff
@@ -32,11 +32,27 @@ func (a *App) CreateGroup(group *model.Group) (*model.Group, *model.AppError) {
 }
 
 func (a *App) UpdateGroup(group *model.Group) (*model.Group, *model.AppError) {
-	return a.Srv().Store.Group().Update(group)
+	newGroup, err := a.Srv().Store.Group().Update(group)
+
+	if err == nil {
+		messageWs := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_RECEIVED_GROUP, "", "", "", nil)
+		messageWs.Add("group", newGroup.ToJson())
+		a.Publish(messageWs)
+	}
+
+	return newGroup, err
 }
 
 func (a *App) DeleteGroup(groupID string) (*model.Group, *model.AppError) {
-	return a.Srv().Store.Group().Delete(groupID)
+	newGroup, err := a.Srv().Store.Group().Delete(groupID)
+
+	if err == nil {
+		messageWs := model.NewWebSocketEvent(model.WEBSOCKET_EVENT_RECEIVED_GROUP, "", "", "", nil)
```

Comment:
Should this be the same same websocket event as the used to update a group?

Reply:
when we remove/delete a group we are actually update its delete_at timestamp, so once can look at it as the update of one of its properties. As we were discussing some time ago, the GetGroup* call(s) return the group if it exists in the table, "removed" or not, so the redux state is populated with both "removed" and not-removed groups. If the redux state would've only contain the non-removed ones, then I think that we would've need to add a specific DELETE event to actually remove the entry from the redux state when it is "removed" from the UserGroups table, with the contract that the redux state contains only non-removed groups.